core/runtime/src/main/java/io/quarkus/runtime/BannerRecorder.java

```diff
@@ -0,0 +1,18 @@
+package io.quarkus.runtime;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import io.quarkus.runtime.annotations.Recorder;
+
+@Recorder
+public class BannerRecorder {
+
+    public RuntimeValue<Optional<Supplier<String>>> provideBannerSupplier(String bannerText,
+            BannerRuntimeConfig bannerRuntimeConfig) {
+        if (bannerRuntimeConfig.enabled) {
+            return new RuntimeValue<>(Optional.of(() -> bannerText));
```

Comment:
Any chance we could avoid the lambda? It will end up being executed at runtime, right?

Reply:
I was pondering doing this anyway... Now I have a motive to 🙃.

Reply:
Done